hearth/spec/hearth/configuration_spec.rb

```diff
@@ -31,49 +24,34 @@ def self.defaults
       it 'uses the config resolver' do
         options = { simple_option: 'test' }
         expect(Hearth::Config::Resolver).to receive(:resolve)
-          .with(an_instance_of(config_class), options, config_class.defaults)
-
-        config_class.new(**options)
+          .with(
+            an_instance_of(config_class),
+            options,
+            { simple_option: ['default'] }
+          ).and_call_original
+
+        config = config_class.new(**options)
+        expect(config.simple_option).to eq('test')
       end
 
       it 'resolves correctly when not passed any options' do
         expect(Hearth::Config::Resolver).to receive(:resolve)
-          .with(an_instance_of(config_class), {}, config_class.defaults)
-
-        config_class.new
-      end
-
-      it 'raises on invalid types' do
-        expect do
-          config_class.new(simple_option: 1)
-        end.to raise_error(
-          ArgumentError,
-          'Expected config[:simple_option] to be in [String], got Integer.'
-        )
-      end
-
-      it 'raises on unknown options' do
-        expect do
-          config_class.new(unknown_option: 'test')
-        end.to raise_error(
-          ArgumentError,
-          'Unexpected members: [config[:unknown_option]]'
-        )
+          .with(
+            an_instance_of(config_class),
+            {},
+            { simple_option: ['default'] }
+          ).and_call_original
+
+        config = config_class.new
+        expect(config.simple_option).to eq('default')
       end
     end
 
-    describe '#dup' do
-      let(:simple_option) { String.new('test') }
-      let(:complex_option) { {} }
-
-      it 'deep copies' do
-        config = config_class.new(
-          simple_option: simple_option,
-          complex_option: complex_option
-        )
-        expect(simple_option).to receive(:dup)
-        expect(complex_option).to receive(:dup)
-        config.dup
+    describe '#merge' do
+      it 'merges the configuration' do
+        config = config_class.new(simple_option: 'test')
+        merged = config.merge(config_class.new(simple_option: 'test2'))
+        expect(merged.simple_option).to eq('test2')
```

Comment:
Should we test invalid options in the merge?

Reply:
Merge doesn't validate.

Reply:
I meant unknown rather than invalid - and I think that may be worth testing as well - I'd consider it a useful part of the interface for users - if they pass something wrong to the options on an operation, I think it should fail rather than silently ignore it.